framework/src/main/java/org/checkerframework/common/aliasing/AliasingVisitor.java

```diff
@@ -271,7 +271,13 @@ private boolean canBeLeaked(Tree exp) {
         AnnotatedTypeMirror type = atypeFactory.getAnnotatedType(exp);
         boolean isMethodInvocation = exp.getKind() == Kind.METHOD_INVOCATION;
         boolean isNewClass = exp.getKind() == Kind.NEW_CLASS;
-        return type.hasExplicitAnnotation(Unique.class) && !isMethodInvocation && !isNewClass;
+        boolean isArray = type.getUnderlyingType().getKind().toString().equalsIgnoreCase("array");
```

Comment:
You should try to avoid operations on strings. Is there no TypeKind to compare against?

Reply:
Ping.

Reply:
I have used your suggested logic in my latest commit. There is no need for array TypeKind comparison in the code.